subsys/bluetooth/host/att.c

```diff
@@ -595,6 +595,13 @@ static void att_on_sent_cb(struct bt_att_tx_meta_data *meta)
 
 	LOG_DBG("opcode 0x%x", meta->opcode);
 
+	if (!meta->att_chan ||
+	    !meta->att_chan->att ||
+	    !meta->att_chan->att->conn) {
```

Comment:
Any chance that `meta->att_chan->att->conn` when the `conn` is e.g. `BT_CONN_DISCONNECTING`?

Reply:
yes but in that case the object is still valid. What do you think, should we still call the cb?

Reply:
Not sure - The function has different behavior depending on the `op_type`, right? 

So if we are sending a GATT notification with the callback set, then that callback is what is called. In that case, I'd still expect that callback. 

If the op_type is request or indication, it seems that we retry those? That shouldn't happen. 

And if I understand `att_sent` correctly, it would attempt to send the next, which also doesn't make sense. 

So at least to me, it seems that it only makes sense to call `chan_req_notif_sent`, and not the other functions in that case.

Reply:
we already eat the cb in the case the ATT chan has been torn down: https://github.com/jori-nordic/zephyr/blob/66a36e84860c11d15a7035f9815a7960a73242be/subsys/bluetooth/host/att.c#L580